### PR TITLE
Update the decline reference journey

### PR DESCRIPTION
**Pull request by davidgisbey**
## Context

Currently, when a referee clicks the link in the email to decline a reference they are taken to a Thank you page. 

If that's a mistake they have to get in touch. We want to enable referees to go into the referee flow and encourage them to think again about giving a reference.

(Plus it's reaaaaal ugly rn)

## Changes proposed in this pull request

- Add radio buttons and additional content to refuse feedback page

|Before|After|
|------------|------------|
| image | image |

- If they choose that they want to provide a reference send them to the confirm relationship page
- Ensure there's a backlink on that page so they can still go back and refuse to give a reference

- If they confirm they don't want to provide a reference redirect them to the thank you page

image

## Guidance to review

This section of the code base is pretty horrible. I'm going to give it a bit of a refactor over Christmas.

## Link to Trello card

https://trello.com/c/IJzqtJI7/2592-%F0%9F%92%942%EF%B8%8F%E2%83%A3-dev-change-the-referee-declining-journey

## Things to check

- [x] This code does not rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-teacher-training#azure-hosting-devops-pipeline)

### CODE REVIEW

File: app/views/referee_interface/reference/refuse_feedback.html.erb [hunk 1]
```diff
@@ -1,21 +1,28 @@
...
     ) do |f| %>
-      <%= f.govuk_submit t('referee.refuse_feedback.confirm') %>
-    <% end %>
+      <%= f.govuk_error_summary %>
+
+      <h1 class="govuk-heading-xl">
+        <%= t('page_titles.referee.refuse_feedback') %>
+      </h1>
+
+      <p class='govuk-body'>Teacher training courses can fill up quickly, and candidates need 2 references to apply.</p>
```
Comment (paulrobertlloyd) [inline app/views/referee_interface/reference/refuse_feedback.html.erb:16]:
```suggestion
      <p class="govuk-body">Teacher training courses can fill up quickly, and candidates need 2 references to apply.</p>
```

File: app/views/referee_interface/reference/refuse_feedback.html.erb [hunk 2]
```diff
@@ -1,21 +1,28 @@
...
 
-    <p class="govuk-body">
-      <%= govuk_link_to t('referee.refuse_feedback.cancel'), referee_interface_reference_feedback_path(token: @token_param) %>
-    </p>
+      <p class='govuk-body'><%= @application.full_name %> will be able to submit the application quicker if you give a reference.</p>
```
Comment (paulrobertlloyd) [inline app/views/referee_interface/reference/refuse_feedback.html.erb:18]:
```suggestion
      <p class="govuk-body"><%= @application.full_name %> will be able to submit the application quicker if you give a reference.</p>
```

File: config/locales/en.yml [hunk 3]
```diff
@@ -134,7 +134,7 @@ en:
     application_feedback: How can we improve %{section} section?
     application_feedback_thank_you: Thank you for your feedback
     referee:
-      refuse_feedback: Are you sure you will not give %{full_name} a reference?
+      refuse_feedback: You’ve declined to give a reference
```
Comment (paulrobertlloyd) [inline config/locales/en.yml:137]: Asked about this on Slack; not sure you have actually declined at this point, so we may want to make the title active instead, something like ‘Decline to give a reference’ or similar

**Review by paulrobertlloyd — CHANGES_REQUESTED** [review]
Submitting the form without choosing an option results in a completely blank page, it does not show any validation message.

**Comment by davidgisbey** [comment]
Thanks for the feedback. Have applied the changes/fixed the error issue.

File: config/locales/application_form.yml [hunk 4]
```diff
@@ -1020,6 +1020,10 @@ en:
             feedback:
               blank: Enter your reference
               too_many_words: Your reference must be %{count} words or fewer
+        referee_interface/refuse_feedback_form:
+          attributes:
+            choice:
+              blank: Choose whether to decline the reference request
```
Comment (paulrobertlloyd) [inline config/locales/application_form.yml:1026]: Could probably over think this, but I think ‘this’ sounds a bit better than ‘the’

```suggestion
              blank: Choose whether to decline this reference request
```

**Review by paulrobertlloyd — APPROVED** [review]
One tiny copy change, and then this is good to 🚢

**Pull request merged**